src/lib/registration/package_search.rb

```diff
@@ -97,5 +98,30 @@ def connect_product(yast_product)
         summary: yast_product.display_name
       )
     end
+
+    # Returns the status for a given package
+    #
+    # Finding out the status of a package requires to ask libzypp through Yast::Pkg.
+    # If there are many results, it might time quite some time. So in order to speed
+    # up the operation, a status cache is built (see #status_map).
+    #
+    # @param pkg_name [String] Package name
+    def status_for(pkg_name)
+      status_map[pkg_name] || :unknown
+    end
+
+    # Returns a memoized status cache
+    #
+    # @return [Hash<String,Symbol>]
+    def status_map
+      return @status_map if @status_map
+      pkgs = Yast::Pkg.Resolvables(
+        { kind: :package }, [:name, :status]
+      )
+      @status_map = pkgs.each_with_object({}) do |pkg, all|
+        next if all.key?(pkg["name"])
```

Comment:
The `Y2Packager::Resolvable` objects provide nicer `.name` and `.status` methods...